src/vmm/src/builder.rs

```diff
@@ -510,21 +510,15 @@ pub fn build_microvm_from_snapshot(
     )?;
 
     #[cfg(target_arch = "x86_64")]
-    // Check if we need to scale the TSC.
-    // We start by checking if the CPU model in the snapshot is
-    // the same as this host's. If they are the same, we don't
-    // need to do anything else.
-    if !is_same_model(&microvm_state.vcpu_states[0].cpuid) {
```

Comment:
The check here can be removed since, if they are the same model, by setting the stc we end up re-set the tsc to the same value and it doesn't change. This check was not needed before this PR.

Reply:
ok, I can agree with the motivation that this check was superfluous.
However, this change should have been isolated cause is not related to this PR.